src/repositories/jobRepository.ts

```diff
@@ -115,7 +115,7 @@ export class JobRepository extends BaseRepository {
 
   async findOneAndUpdateJob(query): Promise<Job | null> {
     const update = { $set: { startTime: new Date(), status: 'inProgress' } };
-    const options = { sort: { priority: -1, createdTime: 1 }, returnNewDocument: true };
+    const options = { sort: { priority: -1, createdTime: 1 }, returnNewDocument: true, includeResultMetadata: true };
```

Comment:
good catch! think this can be a default option for BaseRepository so we don't have to repeat it on each call